### ISSUE TITLE
[iOS] CollectionView has inconsistent width calculation iOS and Android when in a Grid with 'auto' column definition

### ISSUE THREAD

**Issue by Alex-stewart1** [dotnet/maui#21385]
### Description

CollectionView seems to be unable to calculate its required width from its contained items. When a CollectionView is added to a grid where 'auto' is used in the column definition the CollectionView ignores it and stretches off the screen regardless of how much space it actually requires. This behaviour differs between Android and iOS, Android behaves as expected with the CollectionView taking only the space it requires when auto is used

I have tried all permutation's I could think of, adding a second column with a hard coded width etc. Nothing works while 'auto' is used. '*' works as expected

Xaml:

<ContentPage
    x:Class="MauiApp1.MainPage"
    xmlns="http://schemas.microsoft.com/dotnet/2021/maui"
    xmlns:x="http://schemas.microsoft.com/winfx/2009/xaml"
    xmlns:local="clr-namespace:MauiApp1"
    x:DataType="local:MainPage">

    <Grid
        Margin="30"
        ColumnDefinitions="auto"
        RowDefinitions="*">
        <CollectionView
            x:Name="CollectionView"
            BackgroundColor="Yellow"
            ItemsSource="{Binding ImageCollection}">
            <CollectionView.ItemTemplate>
                <DataTemplate>
                    <Label Text="{Binding .}" WidthRequest="100" HeightRequest="50"/>
                </DataTemplate>
            </CollectionView.ItemTemplate>
        </CollectionView>
    </Grid>
</ContentPage>

Code behind file:

```
using System.Collections.ObjectModel;

namespace MauiApp1;
public partial class MainPage : ContentPage
{
    public ObservableCollection<string> ImageCollection { get; } = ["This is some text", "This is more text"];
    public MainPage()
    {
        BindingContext = this;
        InitializeComponent();

    }
}
```

Expected behaviour (Correct on Android):
![AndroidCVExpected](https://github.com/dotnet/maui/assets/113424735/a6b9bc37-bbd7-4a1f-8ef2-c68f57d6ae1a)

Actual behaviour (Incorrect on iOS):
![iOS_CVIncorrect](https://github.com/dotnet/maui/assets/113424735/1dad4691-5ddb-413b-a161-de1e437c1505)

### Steps to Reproduce

1. Create a new default Maui app
2. Add a 'Grid' with a column definition of 'auto' 
3. Add a 'CollectionView' inside the 'auto' column -- (Or use the code provided as an example)
4. Run on iOS

### Version with bug

8.0.10 SR3

### Is this a regression from previous behavior?

Not sure, did not test other versions

### Last version that worked well

Unknown/Other

### Affected platforms

iOS

### Affected platform versions

Only tested on iOS 17.2

### Did you find any workaround?

No not for 'auto'. Switching to '*' or hard coding the column width / CollectionView width fixes, but that can't be done when the CollectionView width needs to be dynamic such as when its content size is unknown ahead of time (like loading images).

**Comment by kevinxufei** [comment]
Verified this issue with Visual Studio 17.10.0 Preview 2 (8.0.14/8.0.7). Can repro on iOS platform.

**Comment by rmarinho** [comment]
Hey @Alex-stewart1  have you tried wrap the label on a Grid for example on the ItemTemplate? 

thanks

**Comment by Alex-stewart1** [comment]
Hey @rmarinho Yes I have tried, the code I showed is a simplified example trying to showcase the issue with as little unrelated code as possible. My actual usage wraps it with a Frame, but it doesn't regardless of whatever is within the ItemTemplate.

**Comment by Alex-stewart1** [comment]
I have added a repo with code to reproduce

**Comment by Alex-stewart1** [comment]
@mariomoramontero Agreed, their is definitely something broken with how the CollectionView calculates it's size when it's ItemTemplate size isn't fixed. This is preventing our company from submitting an iOS app for review. Given the lack of updates it doesn't seem like anything will get done about this anytime soon

**Comment by Alex-stewart1** [comment]
@mariomoramontero This thread is discussing a similar issue to yours https://github.com/dotnet/maui/issues/9135 and more specifically there seems to be a (albeit dodgy) workaround: https://github.com/dotnet/maui/issues/9135
I'm going to try and implement it for my scenario (width not height) and see if I can get around it

**Comment by mariomoramontero** [comment]
> @mariomoramontero This thread is discussing a similar issue to yours #9135 and more specifically there seems to be a workaround: [#9135 (comment)](https://github.com/dotnet/maui/issues/9135#issuecomment-1398493085)
> 
> I'm going to try and implement it for my scenario (width not height) and see if I can get around it
Hi Alex, sadly that workaround from did not work for me, I finally gave a fixed height to the row height instead of "Auto"

**Comment by Alex-stewart1** [comment]
Found a work around by just using a VerticalStackLayout with its BindableLayout.ItemSource bound to for the items. This is fine for small collections of items like in my case, but has lower performance for larger lists (according to the docs)

**Comment by Felicity-R** [comment]
I'm running into this in my app too, and it's still an issue in .Net 9 (confirmed by updating Alex-stewart1's linked github project to .Net 9).
On Android the width is nicely fitted to the labels inside the CollectionView, but on iOS it stretches far too wide.

**Comment by Felicity-R** [comment]
I'm actually not using an Auto sized Grid in my real app, I'm using an AbsoluteLayout with AutoSize for the height/width in the LayoutBounds, but it has the same effect.

**Comment by l4nk4b3l** [comment]
Any news on this? It is still not working ... This should be fixed before ListView gets removed.
